service/history/historyCache.go

```diff
@@ -72,38 +72,65 @@ func newHistoryCache(shard ShardContext) *historyCache {
 	}
 }
 
-func (c *historyCache) getOrCreateWorkflowExecution(domainID string,
-	execution workflow.WorkflowExecution) (workflowExecutionContext, releaseWorkflowExecutionFunc, error) {
-	return c.getOrCreateWorkflowExecutionWithTimeout(context.Background(), domainID, execution)
-}
+func (c *historyCache) getOrCreatCurrentWorkflowExecution(
```

Comment:
this func is mostly the same as getOrCreateWorkflowExecution() except for a flag and another validation call. Consider factoring out the common pieces into a helper method that both of these  can call.

Reply:
sure